### PR TITLE
Declared variables on weather bus

### LINKED ISSUE #1798: Declare variables on weather data bus
The weather data bus has various variables connected to it. These are implicitly defined through the `connect` statement.

In OMEdit, this makes the use of the weather data bus and the weather data reader cumbersome for two reasons:

1. when connecting an input to the bus, a users has to enter the variable name name. E.g., the user has to look up in the weather data reader what the variable is named, and then go back and retype that name. The GUI does not allow to select the name.
2. When opening a model with the weather data reader, OMEdit parses the model and issues an error message because the weather data reader displays `weaBus.TDryBul` (with `DynamicSelect`) but this variable only exists after translation. This is actually a case where the model violates the Modelica Language Standard, so that should be fixed anyway.

Both issues can be addressed by declaring the quantities on the bus in the declaration of the weather data bus.

### PR DESCRIPTION
**Pull request by mwetter**
This closes #1798

### CODE REVIEW

**Comment by mwetter** [comment]
@jelgerjansen : This is ready for review. Can you please review as this will also make the tutorial more practical, see https://github.com/ibpsa/modelica-ibpsa/pull/1799/commits/395265eb35619af19af4816b2cb56d3c963919e2

**Review by jelgerjansen — CHANGES_REQUESTED** [review]
Very nice fix Michael, thanks!
I only have one small comment you can have a look at.
Furthermore, I wasn't able to reproduce the error message you mention [here](https://github.com/ibpsa/modelica-ibpsa/issues/1798#issue-1909644887) when opening the weather data reader itself or a model that contains the reader. When/where did you see this error appearing? (I used version OMEdit v1.21.0)

File: IBPSA/BoundaryConditions/WeatherData/Bus.mo [hunk 1]
```diff
@@ -2,6 +2,44 @@ within IBPSA.BoundaryConditions.WeatherData;
 expandable connector Bus "Data bus that stores weather data"
   extends Modelica.Icons.SignalBus;
 
+  Modelica.Units.SI.Temperature TDryBul(displayUnit="degC")
```
[the remark this answers is not in the document]
Comment (mwetter) [reply IBPSA/BoundaryConditions/WeatherData/Bus.mo]: Thanks, I removed the `displayUnit`

**Comment by mwetter** [comment]
@jelgerjansen : Regarding the error message, this is with 1.22.0~dev-351-g208d900, using `OMEdit --NAPI=true` which activates the new back-end that will soon be the default.

**Review by jelgerjansen — APPROVED** [review]

**Pull request merged**